src/net.cpp

```diff
@@ -852,10 +852,15 @@ void CheckOffsetDisconnectedPeers(const CNetAddr& ip)
         // clear the set
         setOffsetDisconnectedPeers.clear();
         // Trigger the warning
-        std::string strMessage = _("Warning: Peers are being disconnected due time differences. Please check that your computer's date and time are correct! If your clock is wrong PIVX Core will not work properly.");
-        strMiscWarning = strMessage;
-        LogPrintf("*** %s\n", strMessage);
-        uiInterface.ThreadSafeMessageBox(strMessage, "", CClientUIInterface::MSG_ERROR);
+        strMiscWarning = _("Warning: Peers are being disconnected due time differences. Please check that your computer's date and time are correct! If your clock is wrong PIVX Core will not work properly.");
+        LogPrintf("*** %s\n", strMiscWarning);
+
+        static int64_t nLastGUINotif = 0;
+        int64_t now = GetTime();
+        if (nLastGUINotif + 40 < now) { // Do not inform like crazy, the user need to be able to read the dialog.
+            nLastGUINotif = now;
+            uiInterface.ThreadSafeMessageBox(_("Peers are being disconnected due time differences.\n\nPlease check that your computer's date and time are correct! If your clock is wrong PIVX Core will not work properly."), _("Warning"), CClientUIInterface::MSG_ERROR);
+        }
```

Comment:
Do we really want to duplicate this (translated) string entirely? Can't we reuse this version (without `Warning` prefix and with `\n\n` in the middle) also for the logs?

Reply:
I'm against jump lines in the logs, it's harder to read for anyone trying to understand what happened. And adding two of them make it worst (blank space where we don't know what happened, most of the time thinking that it was a bug or something else).

And at the same time, the GUI notification need a jump line to make the text readable.

So.. I think that it's a tiny scenario that will only happen when the internal clock is wrong and only every 40 seconds. It's "ok" to have two separate texts, one for the GUI and one the logging.

Reply:
what about something like this?
It would prevent the translation of two identical lines.
```cpp
std::string strWarn1 = _("Peers are being disconnected due time differences.");
std::string strWarn2 = _("Please check that your computer's date...");

LogPrintf("*** Warning: %s %s\n", strWarn1, strWarn2);

static int64_t nLastGUINotif = 0;
int64_t now = GetTime();
if (nLastGUINotif + 40 < now) {
    nLastGUINotif = now;
    uiInterface.ThreadSafeMessageBox(strprintf("%s\n\n%s", strWarn1, strWarn2), _("Warning"), CClientUIInterface::MSG_ERROR);
}
```
PS: anyway better to avoid updating `strMiscWarning`, as we don't clear it (log lines and gui messageBoxes should be enough).

Reply:
done, updated.